### ISSUE TITLE
ClassCastException: java.lang.String incompatible with java.nio.file.FileTreeWalker$Event

### ISSUE THREAD

**Issue by pshipton** [eclipse-openj9/openj9#8173]
https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_mac_Nightly/235
```
22:49:10  Creating images/jmods/java.scripting.jmod
22:49:11  Creating images/jmods/java.se.jmod
22:49:13  java.lang.ClassCastException: java.lang.String incompatible with java.nio.file.FileTreeWalker$Event
22:49:13  	at java.base/java.nio.file.Files$$Lambda$6/0000000000000000.test(Unknown Source)
22:49:13  	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
22:49:13  	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
22:49:13  	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
22:49:13  	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:497)
22:49:13  	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:487)
22:49:13  	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
22:49:13  	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
22:49:13  	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:239)
22:49:13  	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
22:49:13  	at java.base/jdk.internal.module.IllegalAccessMaps.generate(IllegalAccessMaps.java:96)
22:49:13  	at java.base/jdk.internal.module.ModuleBootstrap.addIllegalAccess(ModuleBootstrap.java:784)
22:49:13  	at java.base/jdk.internal.module.ModuleBootstrap.boot(ModuleBootstrap.java:433)
22:49:13  	at java.base/java.lang.ClassLoader.initializeClassLoaders(ClassLoader.java:211)
22:49:13  	at java.base/java.lang.Thread.initialize(Thread.java:430)
22:49:13  	at java.base/java.lang.Thread.<init>(Thread.java:155)
```

Restarted the build, and the restart built correctly.
https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_mac_Nightly/236/

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_mac_Nightly/250

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDKnext_x86-64_linux_OpenJDK/135

**Comment by DanHeidinga** [comment]
How reproducible is this?  Think we could capture a core with the right -Xdump option?

**Comment by smlambert** [comment]
You can append command line options to any test using EXTRA_OPTIONS parameter... (you may need to \"-Xdump:xxxxx\" if the option contains lots of special characters.
<img width="548" alt="Screen Shot 2020-01-20 at 12 26 06 PM" src="https://user-images.githubusercontent.com/2836948/72746416-1147eb00-3b80-11ea-8015-92ab81049742.png">


if that is not what you want to do, or the options use a lot of complex characters that the framework chokes on, you can set an environment variable in a branch of openjdk-tests and run material from that branch ([how to pass environment vars](https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/doc/userGuide.md#how-to-pass-in-environment-variables))

**Comment by pshipton** [comment]
My bad, running a grinder isn't helpful for this one. It's actually the build (compiling the jvm) that needs the environment variable set. @AdamBrousseau how can this be done?

**Comment by AdamBrousseau** [comment]
Could add it to variables file to test in a PR build.
https://github.com/eclipse/openj9/blob/master/buildenv/jenkins/variables/defaults.yml#L385

Alternatively, could do it in a personal build according to [this line](https://github.com/eclipse/openj9/blob/master/buildenv/jenkins/common/variables-functions.groovy#L587) you can pass param `BUILD_ENV_VARS` to top level build.

**Comment by pshipton** [comment]
Created #8348 to add the -Xdump option to osx and xlinux builds.

**Comment by pshipton** [comment]
@keithc-ca is also trying to recreate the problem locally by building over and over, removing the jmods and image directory and the shared cache between every run.

**Comment by keithc-ca** [comment]
It finally happened for me locally on xlinux (with cmake, not that that likely matters), but I didn't have the environment variable set to get dumps. I guess all that says is that the problem is not yet fixed.

**Comment by pshipton** [comment]
Perhaps it's made a comeback, we just caught one here https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_mac_Personal/509
@DanHeidinga someone should take a look while the diagnostics are still available.

**Comment by pshipton** [comment]
Another occurrence https://ci.eclipse.org/openj9/job/Build_JDKnext_x86-64_mac_OpenJDK/224
There is a core file created, available in the diagnostics download https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDKnext_x86-64_mac_OpenJDK/224/Build_JDKnext_x86-64_mac_OpenJDK-224-20200711-022921-diagnostics.tar.gz

**Comment by adamfarley** [comment]
Another instance:
https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-mac-x64-openj9/742/console

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_linux_xl_Release/50
```
08:02:31  Creating images/jmods/java.rmi.jmod
08:02:31  Creating images/jmods/java.scripting.jmod
08:02:31  JVMDUMP039I Processing dump event "systhrow", detail "java/lang/ClassCastException" at 2020/10/17 08:02:25 - please wait.
08:02:31  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Build_JDK11_x86-64_linux_xl_Release/make/core.20201017.080225.31735.0001.dmp' in response to an event
08:02:31  Creating images/jmods/java.se.jmod
08:02:31  Creating images/jmods/java.security.jgss.jmod
08:02:38  Creating images/jmods/java.security.sasl.jmod
08:02:38  JVMDUMP010I System dump written to /home/jenkins/workspace/Build_JDK11_x86-64_linux_xl_Release/make/core.20201017.080225.31735.0001.dmp
08:02:38  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Build_JDK11_x86-64_linux_xl_Release/make/javacore.20201017.080225.31735.0002.txt' in response to an event
08:02:46  Creating images/jmods/java.smartcardio.jmod
08:03:04  JVMDUMP010I Java dump written to /home/jenkins/workspace/Build_JDK11_x86-64_linux_xl_Release/make/javacore.20201017.080225.31735.0002.txt
08:03:04  JVMDUMP013I Processed dump event "systhrow", detail "java/lang/ClassCastException".
08:03:04  java.lang.ClassCastException: java.lang.String incompatible with java.nio.file.FileTreeWalker$Event
```
```
1XMCURTHDINFO  Current thread
3XMTHREADINFO      "main" J9VMThread:0x00002B6D4C0C7400, omrthread_t:0x00002B6D4C01BD50, java/lang/Thread:0x00002B6D77B6BA50, state:R, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x1, isDaemon:false)
3XMTHREADINFO1            (native thread ID:0x7BFC, native priority:0x5, native policy:UNKNOWN, vmstate:R, vm thread flags:0x00001020)
3XMTHREADINFO2            (native stack address range from:0x00002B6D486E2000, to:0x00002B6D488E3000, size:0x201000)
3XMCPUTIME               CPU usage total: 1.115526795 secs, current category="System-JVM"
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at java/nio/file/Files$$Lambda$6/000000000000000000.test(Bytecode PC:5(Compiled Code))
4XESTACKTRACE                at java/util/stream/ReferencePipeline$2$1.accept(ReferencePipeline.java:176(Compiled Code))
4XESTACKTRACE                at java/util/Iterator.forEachRemaining(Iterator.java:133)
4XESTACKTRACE                at java/util/Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
4XESTACKTRACE                at java/util/stream/AbstractPipeline.copyInto(AbstractPipeline.java:497)
4XESTACKTRACE                at java/util/stream/AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:487)
4XESTACKTRACE                at java/util/stream/ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
4XESTACKTRACE                at java/util/stream/ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
4XESTACKTRACE                at java/util/stream/AbstractPipeline.evaluate(AbstractPipeline.java:239)
4XESTACKTRACE                at java/util/stream/ReferencePipeline.forEach(ReferencePipeline.java:497)
4XESTACKTRACE                at jdk/internal/module/IllegalAccessMaps.generate(IllegalAccessMaps.java:96)
4XESTACKTRACE                at jdk/internal/module/ModuleBootstrap.addIllegalAccess(ModuleBootstrap.java:784)
4XESTACKTRACE                at jdk/internal/module/ModuleBootstrap.boot(ModuleBootstrap.java:433)
4XESTACKTRACE                at java/lang/ClassLoader.initializeClassLoaders(ClassLoader.java:211)
4XESTACKTRACE                at java/lang/Thread.initialize(Thread.java:430)
4XESTACKTRACE                at java/lang/Thread.<init>(Thread.java:155)
```

**Comment by pshipton** [comment]
@andrewcraik we got another occurrence of this, with a core file. Can somebody take a look pls, although I'm not 100% sure this is a JIT problem.

https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDK11_x86-64_linux_xl_Release/50/Build_JDK11_x86-64_linux_xl_Release-50-20201017-080312-diagnostics.tar.gz

**Comment by andrewcraik** [comment]
@liqunl given this is related to 292 could you or @nbhuiyan have a look - note the diagnostic files are available so if you could look before they disappear that would be great

**Comment by liqunl** [comment]
@nbhuiyan is looking at the issue. For the record, the sha levels of the failed job at https://github.com/eclipse/openj9/issues/8173#issuecomment-712859275 is
```
OpenJ9: 5bc5d63
OMR: 582366a
OpenJDK: 5e617ab
```

**Comment by nbhuiyan** [comment]
This exception was thrown because `java/nio/file/Files$$Lambda$6/000000000000000000.test(Ljava/lang/Object;)Z` received an argument of type `String` instead of something that would pass `checkcast` against `java.nio.file.FileTreeWalker$Event`. @liqunl and I attempted tracing the origin of this `String` object and determining its cause, but it originates from a point that is too far from the exception site for us to get an answer from this coredump alone. This failure did not show up in any of the 30 builds I attempted locally.

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdknext_j9_sanity.functional_ppc64le_linux_xl_OpenJDK_testList_1/88/
Diagnostics https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDKnext_x86-64_mac_OpenJDK/339/Build_JDKnext_x86-64_mac_OpenJDK-339-20201126-003927-diagnostics.tar.gz
```
00:39:01  Generating javadoc for openj9 source files
00:39:01  Optimizing the exploded image
00:39:01  Creating java.compiler.jmod
00:39:01  Creating java.datatransfer.jmod
00:39:01  Creating java.instrument.jmod
00:39:01  Creating java.logging.jmod
00:39:01  Creating java.management.jmod
00:39:01  Creating java.management.rmi.jmod
00:39:01  Creating java.desktop.jmod
00:39:01  Creating java.naming.jmod
00:39:01  Creating java.net.http.jmod
00:39:09  JVMDUMP039I Processing dump event "systhrow", detail "java/lang/ClassCastException" at 2020/11/26 00:39:05 - please wait.
```
`Detail "java/lang/ClassCastException" "java.lang.String incompatible with java.nio.file.FileTreeWalker$Event" received`
```
3XMTHREADINFO      "(unnamed thread)" J9VMThread:0x00000000136DDC00, omrthread_t:0x00007FC7BE003450, java/lang/Thread:0x00000000FF806E20, state:R, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x1, isDaemon:false)
3XMTHREADINFO1            (native thread ID:0x72CD608, native priority:0x5, native policy:UNKNOWN, vmstate:R, vm thread flags:0x00001020)
3XMTHREADINFO2            (native stack address range from:0x0000700007B01000, to:0x0000700007B81000, size:0x80000)
3XMCPUTIME               CPU usage total: 0.611429000 secs, current category="System-JVM"
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at java/nio/file/Files$$Lambda$6/0x0000000000000000.test(Bytecode PC:5(Compiled Code))
4XESTACKTRACE                at java/util/stream/ReferencePipeline$2$1.accept(ReferencePipeline.java:176(Compiled Code))
4XESTACKTRACE                at java/util/Iterator.forEachRemaining(Iterator.java:133)
4XESTACKTRACE                at java/util/Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
4XESTACKTRACE                at java/util/stream/AbstractPipeline.copyInto(AbstractPipeline.java:497)
4XESTACKTRACE                at java/util/stream/AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:487)
4XESTACKTRACE                at java/util/stream/ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
4XESTACKTRACE                at java/util/stream/ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
4XESTACKTRACE                at java/util/stream/AbstractPipeline.evaluate(AbstractPipeline.java:239)
4XESTACKTRACE                at java/util/stream/ReferencePipeline.forEach(ReferencePipeline.java:594)
4XESTACKTRACE                at jdk/internal/module/IllegalAccessMaps.generate(IllegalAccessMaps.java:98)
4XESTACKTRACE                at jdk/internal/module/ModuleBootstrap.addIllegalAccess(ModuleBootstrap.java:842)
4XESTACKTRACE                at jdk/internal/module/ModuleBootstrap.boot2(ModuleBootstrap.java:483)
4XESTACKTRACE                at jdk/internal/module/ModuleBootstrap.boot(ModuleBootstrap.java:188)
4XESTACKTRACE                at java/lang/ClassLoader.initializeClassLoaders(ClassLoader.java:215)
4XESTACKTRACE                at java/lang/Thread.initialize(Thread.java:444)
4XESTACKTRACE                at java/lang/Thread.<init>(Thread.java:161)
```

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK11_s390x_linux_xl_OpenJDK11/48
No diagnostics
```
[2020-12-03T06:47:12.987Z] Creating support/test/jdk/jtreg/native/lib/libsanity_SimpleNativeLib2.so from 1 file(s)
[2020-12-03T06:47:12.987Z] Creating support/test/jdk/jtreg/native/bin/sanity_SimpleNativeLauncher from 1 file(s)
[2020-12-03T06:47:12.987Z] Creating support/test/jdk/jtreg/native/bin/JliLaunchTest from 1 file(s)
[2020-12-03T06:47:14.674Z] Note: Some input files use or override a deprecated API.
[2020-12-03T06:47:14.674Z] Note: Recompile with -Xlint:deprecation for details.
[2020-12-03T06:47:14.674Z] Note: Some input files use unchecked or unsafe operations.
[2020-12-03T06:47:14.674Z] Note: Recompile with -Xlint:unchecked for details.
[2020-12-03T06:47:14.674Z] Creating support/modules_cmds/openj9.dtfj/jextract from 1 file(s)
[2020-12-03T06:47:16.131Z] Generating javadoc for openj9 source files
[2020-12-03T06:47:38.115Z] Creating destination directory: "/home/jenkins/workspace/Build_JDK11_s390x_linux_xl_OpenJDK11/build/linux-s390x-normal-server-release/images/openj9-docs/api/"
```

**Comment by nbhuiyan** [comment]
Same story using the diagnostics from the [macOS build failure on 26 Nov 2020](https://ci.eclipse.org/openj9/job/Build_JDKnext_x86-64_mac_OpenJDK/339/) as [comment above](https://github.com/eclipse/openj9/issues/8173#issuecomment-715604692). Furthermore, I have not been able to reproduce this failure locally on macOS. Therefore, we do not have a way to resolve this issue as part of 0.24.

**Comment by 0xdaryl** [comment]
Lack of reproducibility and useful diagnostic info make this unlikely for 0.24.  I would like to see us get "creative" in attempting to collect diagnostic info for 0.25.

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK15_s390x_linux_xl_Release/13

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK15_x86-64_linux_Nightly/175
```
[2021-01-19T04:09:25.346Z] java.lang.ClassCastException: java.lang.String incompatible with java.nio.file.FileTreeWalker$Event
[2021-01-19T04:09:25.346Z] JVMDUMP010I Java dump written to /home/jenkins/workspace/Build_JDK15_x86-64_linux_Nightly/make/javacore.20210119.000909.30825.0002.txt
```

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_mac_OMR/818
```
16:26:57  JVMDUMP039I Processing dump event "systhrow", detail "java/lang/ClassCastException" at 2021/01/29 16:26:57 - please wait.
16:26:57  JVMDUMP032I JVM requested System dump using '/Users/jenkins/workspace/Build_JDK11_x86-64_mac_OMR/make/core.20210129.162657.25346.0001.dmp' in response to an event
16:26:59  JVMDUMP010I System dump written to /Users/jenkins/workspace/Build_JDK11_x86-64_mac_OMR/make/core.20210129.162657.25346.0001.dmp
16:26:59  JVMDUMP032I JVM requested Java dump using '/Users/jenkins/workspace/Build_JDK11_x86-64_mac_OMR/make/javacore.20210129.162657.25346.0002.txt' in response to an event
16:26:59  JVMDUMP010I Java dump written to /Users/jenkins/workspace/Build_JDK11_x86-64_mac_OMR/make/javacore.20210129.162657.25346.0002.txt
16:26:59  JVMDUMP013I Processed dump event "systhrow", detail "java/lang/ClassCastException".
16:26:59  java.lang.ClassCastException: java.lang.String incompatible with java.nio.file.FileTreeWalker$Event
```

**Comment by 0xdaryl** [comment]
Little progress was made on this despite having a new crash dump file to work with.  Moving out one more release.

**Comment by 0xdaryl** [comment]
Unlikely to be fixed for 0.26 again.  Moving out.

**Comment by pshipton** [comment]
Haven't seen this for some time, closing until seen again.
